manifests/init.pp

```diff
@@ -40,6 +40,21 @@
   Boolean $iptables_docker_chain = simplib::lookup('simp_options::firewall', { 'default_value' => false }),
 ) {
 
+  # Need to account for changing the docker_group in one of the options hashes
+  if $options and $options['docker_group'] and !$options['socket_group'] {
+    $_socket_group_option = {
+      'socket_group' => $options['docker_group']
+    }
+  }
+  elsif $default_options[$release_type] and $default_options[$release_type]['docker_group']  and !$default_options[$release_type]['socket_group'] {
+    $_socket_group_option = {
+      'socket_group' => $default_options[$release_type]['docker_group']
+    }
+  }
+  else {
+    $_socket_group_option = {}
+  }
+
```

Comment:
I don't think this kind of logic is required. The upstream module makes it a different parameter, so if this is a real bug where users need to set the socket group, let them set the socket group in the options hash. We could even set the default in the default_options param.

Reply:
@jeefberkey I started by setting the default in `default_options` and then realized that it didn't make logical sense to the user why one would change and the other would not.

Using `alias` inside of the data did not appear to work, thus the logic.

Also, if we want it to 'work as expected', we need to mirror the behavior in both `options` and `default_options`.

Otherwise, we might as well just tell users to set both of them in Hiera and let it remain confusing from the inside out.

Reply:
I agree with @jeefberkey's goal to permit users as much configuration as is feasible, but not the suggestion to "let it remain confusing from the inside out."

I also agree with @trevor-vaughan's reasoning―if further logic is needed to make components work as expected, that's a profile's job.

The suggestion to delegate everything to Hiera would [break the profile](https://travis-ci.org/op-ct/pupmod-simp-simp_docker/jobs/324740593#L1048) and subvert its main purpose, which is "to get Docker up and running on SIMP systems."

I don't see why `docker::socket_group` and `docker::docker_group` would ever be different―docker won't start unless the socket file can be managed by the process, so the use case for that is certainly more obvious.